src/main/webapp/app/shared/statistics-graph/statistics-average-score-graph.component.ts

```diff
@@ -347,6 +354,48 @@ export class StatisticsAverageScoreGraphComponent extends ChartExerciseTypeFilte
         this.exerciseScoresFilteredByPerformanceInterval = this.filterForPerformanceInterval(interval);
     }
 
+    /**
+     * Wrapper method that handles the toggling of a category and sets up the chart accordingly
+     * @param category the category the user selects or deselects
+     */
+    toggleCategory(category: string): void {
+        const filteredAgainstCategory = this.categoryFilter.toggleCategory(this.exerciseScoresFilteredByPerformanceInterval, category) as CourseManagementStatisticsModel[];
```

Comment:
You can avoid those casts if you check how we did this in `user-management.component.ts` (e.g. generics provided via function)

Reply:
Changed in 30b9ea5